pkg/kubectl/cmd/run.go

```diff
@@ -276,14 +276,14 @@ func RunRun(f cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *c
 
 	params["env"] = cmdutil.GetFlagStringArray(cmd, "env")
 
-	allErrs := []error{}
-	var runObjectMap = map[string]*RunObject{}
+	var createdObjects = []*RunObject{}
 	runObject, err := createGeneratedObject(f, cmd, generator, names, params, cmdutil.GetFlagString(cmd, "overrides"), namespace)
 	if err != nil {
-		allErrs = append(allErrs, err)
+		return err
```

Comment:
You should do
```
allErrs = append(allErrs, err)
utilerrors.NewAggregate(allErrs)
```
You don't want to lose earlier error info.

Reply:
I moved the allErrs variable after the ‘if’ statement. Because this error can be returned immediately, there’s no need to use the error aggregator.